metrics-operator/converter/slo_converter.go

```diff
@@ -163,30 +163,132 @@ func setupTarget(o *Objective) (*metricsapi.Target, error) {
+	if err != nil {
+		return target, err
+	}
+	if (len(o.Pass[0].Operators) == 1 && len(o.Warning[0].Operators) == 1) || isWarningSuperInterval {
+		op1, err := newOperator(o.Pass[0].Operators, true)
+		if err != nil {
+			return target, err
 		}
-		if len(o.Warning[0].Operators) > 0 {
-			// TODO cover use cases with multiple operators (create new objectives)
-			op, err := newOperator(o.Warning[0].Operators[0])
-			if err != nil {
-				return target, err
-			}
-			target.Failure = op
+		op2, err := newOperator(o.Warning[0].Operators, true)
+		if err != nil {
+			return target, err
+		}
+		target.Failure = op2
+		target.Warning = op1
+		return target, nil
+	}
+
+	// if pass is superinterval of warn, the following logic is used:
+	// !(pass criteria) -> fail criteria
+	// warn criteria -> warn criteria
+	isPassSuperInterval, err := isSuperInterval(o.Pass[0].Operators, o.Warning[0].Operators)
+	if err != nil {
+		return target, err
+	}
+	if isPassSuperInterval {
+		op1, err := newOperator(o.Warning[0].Operators, false)
+		if err != nil {
+			return target, err
 		}
+		op2, err := newOperator(o.Pass[0].Operators, true)
+		if err != nil {
+			return target, err
+		}
+		target.Failure = op2
+		target.Warning = op1
+		return target, nil
+	}
+
+	return target, nil
+}
+
+// checks if interval is valid and if the first set of operators defines interval
+// which is superset of the interval defined by second set of operators
+func isSuperInterval(op1 []string, op2 []string) (bool, error) {
+	superInterval, err := createInterval(op1)
+	if err != nil {
+		return false, err
+	}
+	subInterval, err := createInterval(op2)
+	if err != nil {
+		return false, err
+	}
+
+	return superInterval.Start.Cmp(subInterval.Start) < 1 && superInterval.End.Cmp(subInterval.End) >= 0, nil
+}
+
+// creates interval from set of operators
+func createInterval(op []string) (*Interval, error) {
+	// if it's unbounded interval, we have only one operator
+	if len(op) == 1 {
+		return createUnboundedInterval(op[0])
 	}
 
-	return target, err
+	//bounded interval
+	return createBoundedInterval(op)
+}
+
+func createBoundedInterval(op []string) (*Interval, error) {
+	if len(op) < 2 {
+		return nil, fmt.Errorf(UnsupportedIntervalCombinationErrMsg, op)
+	}
+	//fetch operators and values
+	operator1, value1, err := decodeOperatorAndValue(op[0])
+	if err != nil {
+		return nil, err
+	}
+	operator2, value2, err := decodeOperatorAndValue(op[1])
+	if err != nil {
+		return nil, err
+	}
+	// determine lower and higher bouds
+	smallerOperator, biggerOperator, err := decideIntervalBounds(operator1, value1, operator2, value2)
+	if err != nil {
+		return nil, err
+	}
+	//check if the interval makes logical sense for conversions, e.g. 5 < x < 10; unsupported: x < 5 && x > 10
+	if (smallerOperator.Operation == ">" || smallerOperator.Operation == ">=") && (biggerOperator.Operation == "<" || biggerOperator.Operation == "<=") {
```

Comment:
maybe worth extracting a function `isLessOrEqual(string)` since the check for < and <= occurs in multiple places
Similarly for > and >=